.github/workflows/test.yml

```diff
@@ -7,59 +7,140 @@ on:
-    runs-on: ${{ matrix.os }}
+  cpp-linux:
+    name: 'C++ tests (gcc-clang/Linux/x64)'
+    runs-on: ubuntu-latest
     strategy:
       fail-fast: false
       matrix:
-        os:
-        - ubuntu-latest
-        - macos-latest
-        - windows-latest
+        build_type:
+          - Debug
+          - Release
+        compiler:
+          - clang++
+          - g++
     steps:
-    - uses: actions/checkout@v2
-      with:
-        submodules: recursive
-    - name: Install dependencies
-      if: matrix.os == 'macos-latest'
-      run: |
-        brew install bison flex
-        echo "/usr/local/opt/bison/bin" >> $GITHUB_PATH
-        echo "/usr/local/opt/flex/bin" >> $GITHUB_PATH
-    - uses: actions/cache@v2
-      if: matrix.os == 'windows-latest'
-      with:
-        path: C:\Users\runneradmin\AppData\Local\Temp\chocolatey
-        key: ${{ runner.os }}-chocolatey-cpp-0
-        restore-keys: |
-          ${{ runner.os }}-chocolatey-cpp-
-          ${{ runner.os }}-chocolatey-
-    - name: Install dependencies
-      if: matrix.os == 'windows-latest'
-      run: choco install winflexbison3 --version 2.5.18.20190508
-    - name: Configure
-      run: cmake . -DCMAKE_BUILD_TYPE=Debug -DOPENQL_BUILD_TESTS=ON -DBUILD_SHARED_LIBS=OFF -DWITH_UNITARY_DECOMPOSITION=OFF
-    - name: Build
-      run: cmake --build . --parallel 5
-    - name: Test
-      run: ctest -C Debug --output-on-failure
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Get latest CMake
+        uses: lukka/get-cmake@latest
+      - name: Create build directory
+        run: mkdir -p build-${{ matrix.compiler }}-${{ matrix.build_type }}
+      - name: Configure
+        env:
+          CXX: ${{ matrix.compiler }}
+        working-directory: build-${{ matrix.compiler }}-${{ matrix.build_type }}
+        run: cmake -S .. -B . -DCMAKE_BUILD_TYPE=${{ matrix.build_type }} -DBUILD_SHARED_LIBS=OFF -DOPENQL_BUILD_TESTS=ON -DWITH_UNITARY_DECOMPOSITION=OFF
+      - name: Build
+        working-directory: build-${{ matrix.compiler }}-${{ matrix.build_type }}
+        run: cmake --build . --parallel 10
+      - name: Test
+        working-directory: build-${{ matrix.compiler }}-${{ matrix.build_type }}
+        run: ctest -C ${{ matrix.build_type }} --output-on-failure
 
-  cpp-standalone:
-    name: 'C++ standalone program'
-    runs-on: ubuntu-latest
+  cpp-macos:
+    name: 'C++ tests (clang/MacOS/x64)'
+    runs-on: macos-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        build_type:
+          - Debug
+          - Release
     steps:
-    - uses: actions/checkout@v2
-      with:
-        submodules: recursive
-    - name: Configure
-      run: |
-        mkdir cbuild
-        cd cbuild && cmake ../examples/cpp-standalone-example
-    - name: Build
-      run: make -C cbuild -j 5
-    - name: Test
-      run: cd tests && ../cbuild/example
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
```

Comment:
fetch-depth

Reply:
Also think of updating assets.yml

Reply:
Done! Well spotted!

I haven't tested `assets.yml` for none of my last changes in libqasm, OpenQL, and QX-simulator. I guess we'll address possible issues there when we have to cross those bridges.